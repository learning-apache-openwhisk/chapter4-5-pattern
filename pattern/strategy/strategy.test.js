const name = require("./name.js").main;
const email = require("./email.js").main;
const phone = require("./phone.js").main;

test("strategy-name", () => {
  expect(name({ name: "Michele" })).toMatchSnapshot();
  expect(name({})).toMatchSnapshot();
  expect(name({})).toMatchSnapshot();
});

test("strategy-email", () => {
  expect(email({ email: "michele@sciabarra.com" })).toMatchSnapshot();
  expect(email({ email: "michele.sciabarra.com" })).toMatchSnapshot();
  expect(email({ email: "" })).toMatchSnapshot();
  expect(email({})).toMatchSnapshot();
});

test("strategy-phone", () => {
  expect(phone({ phone: "123" })).toMatchSnapshot();
  expect(phone({ phone: "(123) 456-7890" })).toMatchSnapshot();
  expect(phone({ phone: "pippolo plutolo" })).toMatchSnapshot();
  expect(phone({ phone: "" })).toMatchSnapshot();
  expect(phone({})).toMatchSnapshot();
});

test("strategy-pipeline", () => {
  expect(name(email(phone({})))).toMatchSnapshot();
  expect(
    name(email(phone({ phone: "123", email: "michele" })))
  ).toMatchSnapshot();
  expect(
    name(
      email(
        phone({
          name: "Michele",
          phone: "1234567890",
          email: "michele@sciabarra.com"
        })
      )
    )
  ).toMatchSnapshot();
});
