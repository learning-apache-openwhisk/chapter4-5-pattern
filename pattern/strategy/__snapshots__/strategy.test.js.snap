// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`strategy-email 1`] = `
Object {
  "email": "michele@sciabarra.com",
  "errors": Array [],
  "message": Array [
    "email: michele@sciabarra.com",
  ],
}
`;

exports[`strategy-email 2`] = `
Object {
  "email": "michele.sciabarra.com",
  "errors": Array [
    "michele.sciabarra.com does not look like an email",
  ],
  "message": Array [],
}
`;

exports[`strategy-email 3`] = `
Object {
  "email": "",
  "errors": Array [
    "missing email",
  ],
  "message": Array [],
}
`;

exports[`strategy-email 4`] = `
Object {
  "errors": Array [
    "missing email",
  ],
  "message": Array [],
}
`;

exports[`strategy-name 1`] = `
Object {
  "errors": Array [],
  "message": Array [
    "name: Michele",
  ],
  "name": "Michele",
}
`;

exports[`strategy-name 2`] = `
Object {
  "errors": Array [
    "missing name",
  ],
  "message": Array [],
}
`;

exports[`strategy-name 3`] = `
Object {
  "errors": Array [
    "missing name",
  ],
  "message": Array [],
}
`;

exports[`strategy-phone 1`] = `
Object {
  "errors": Array [
    "123 does not look like a phone number",
  ],
  "message": Array [],
  "phone": "123",
}
`;

exports[`strategy-phone 2`] = `
Object {
  "errors": Array [],
  "message": Array [
    "phone: (123) 456-7890",
  ],
  "phone": "(123) 456-7890",
}
`;

exports[`strategy-phone 3`] = `
Object {
  "errors": Array [
    "pippolo plutolo does not look like a phone number",
  ],
  "message": Array [],
  "phone": "pippolo plutolo",
}
`;

exports[`strategy-phone 4`] = `
Object {
  "errors": Array [
    "missing phone",
  ],
  "message": Array [],
  "phone": "",
}
`;

exports[`strategy-phone 5`] = `
Object {
  "errors": Array [
    "missing phone",
  ],
  "message": Array [],
}
`;

exports[`strategy-pipeline 1`] = `
Object {
  "errors": Array [
    "missing phone",
    "missing email",
    "missing name",
  ],
  "message": Array [],
}
`;

exports[`strategy-pipeline 2`] = `
Object {
  "email": "michele",
  "errors": Array [
    "123 does not look like a phone number",
    "michele does not look like an email",
    "missing name",
  ],
  "message": Array [],
  "phone": "123",
}
`;

exports[`strategy-pipeline 3`] = `
Object {
  "email": "michele@sciabarra.com",
  "errors": Array [],
  "message": Array [
    "phone: 1234567890",
    "email: michele@sciabarra.com",
    "name: Michele",
  ],
  "name": "Michele",
  "phone": "1234567890",
}
`;
